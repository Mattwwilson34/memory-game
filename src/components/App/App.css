.CardContainer {
  margin-top: 20px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(2, 1fr);
  row-gap: 20px;
  justify-items: center;
}
