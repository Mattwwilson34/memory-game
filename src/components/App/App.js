import './App.css';
// import React, { useState, useEffect } from 'react';
import Header from '../Header/Header';
import Card from '../Card/Card';

function App() {
  return (
    <div className='App' data-testid='App'>
      <Header />
      <Card />
    </div>
  );
}

export default App;
