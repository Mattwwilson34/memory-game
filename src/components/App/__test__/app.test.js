import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../App';

import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import TestRenderer from 'react-test-renderer';
afterAll(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders App correctly', () => {
  render(<App />);
  expect(screen.getByTestId('App')).toHaveTextContent('Test');
});

it('matches snapshot', () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
