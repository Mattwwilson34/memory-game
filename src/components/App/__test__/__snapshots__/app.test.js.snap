// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="App"
>
  Test
</div>
`;
